ndc-models/src/relational_query/expression.rs

```diff
@@ -0,0 +1,921 @@
+    /// * During projection: `relational_query.project.expression.scalar.to_upper`
+    /// * During filtering: `relational_query.filter.scalar.to_upper`
+    /// * During sorting:`relational_query.sort.expression.scalar.to_upper`
+    /// * During joining: `relational_query.join.expression.scalar.to_upper`
+    /// * During aggregation: `relational_query.aggregate.expression.scalar.to_upper`
+    /// * During windowing: `relational_query.window.expression.scalar.to_upper`
+    ToUpper {
+        expr: Box<RelationalExpression>,
+    },
+
+    // acos
+    // acosh
+    // ascii
+    // asin
+    // asinh
+    // atan
+    // atan2
+    // atanh
+    // bit_length
+    // btrim
+    // cbrt
+    // chr
+    // coalesce
+    // concat_ws
+    // contains
+    // cosh
+    // cot
+    // decode
+    // degrees
+    // digest
+    // encode
+    // ends_with
+    // factorial
+    // find_in_set
+    // gcd
+    // initcap
+    // instr
+    // lcm
+    // levenshtein
+    // make_date
+    // md5
+    // nanvl
+    // nvl2
+    // octet_length
+    // pi
+    // radians
+    // regexp_count
+    // regexp_like
+    // regexp_match
+    // regexp_replace
+    // repeat
+    // sha224
+    // sha256
+    // sha384
+    // sha512
+    // signum
+    // sin
+    // sinh
+    // split_part
+    // starts_with
+    // tanh
+    // to_char
+    // to_hex
+    // to_timestamp_micros
+    // to_timestamp_millis
+    // to_timestamp_nanos
+    // to_timestamp_seconds
+    // today
+    // translate
+    // uuid
+
+    // Aggregate functions
+    Average {
+        expr: Box<RelationalExpression>,
+    },
+    /// Only used when in specific contexts where the appropriate capability is supported:
+    /// * During projection: `relational_query.project.expression.aggregate.bool_and`
+    /// * During filtering: `relational_query.filter.aggregate.bool_and`
+    /// * During sorting:`relational_query.sort.expression.aggregate.bool_and`
+    /// * During joining: `relational_query.join.expression.aggregate.bool_and`
+    /// * During aggregation: `relational_query.aggregate.expression.aggregate.bool_and`
+    /// * During windowing: `relational_query.window.expression.aggregate.bool_and`
+    BoolAnd {
+        expr: Box<RelationalExpression>,
+    },
+    /// Only used when in specific contexts where the appropriate capability is supported:
+    /// * During projection: `relational_query.project.expression.aggregate.bool_or`
+    /// * During filtering: `relational_query.filter.aggregate.bool_or`
+    /// * During sorting:`relational_query.sort.expression.aggregate.bool_or`
+    /// * During joining: `relational_query.join.expression.aggregate.bool_or`
+    /// * During aggregation: `relational_query.aggregate.expression.aggregate.bool_or`
+    /// * During windowing: `relational_query.window.expression.aggregate.bool_or`
+    BoolOr {
+        expr: Box<RelationalExpression>,
+    },
+    Count {
```

Comment:
I added `distinct: bool` here in response to a request from @tirumaraiselvan. I didn't add it for all of the other aggregate functions, but we could (although it's rarer in my experience). It would probably need another sub-capability though.

Reply:
Makes sense, `distinct` added to match previous changes and `count_distinct` aggregation capability added.